packages/suite/src/actions/wallet/stakeActions.ts

```diff
@@ -0,0 +1,213 @@
+import BigNumber from 'bignumber.js';
+
+import TrezorConnect, { SignedTransaction } from '@trezor/connect';
+import { getEthereumDefinitions } from '@trezor/connect/src/api/ethereum/ethereumDefinitions';
+import {
+    selectDevice,
+    replaceTransactionThunk,
+    syncAccountsWithBlockchainThunk,
+    stakeActions,
+} from '@suite-common/wallet-core';
+import { notificationsActions } from '@suite-common/toast-notifications';
+import { formatAmount, formatNetworkAmount } from '@suite-common/wallet-utils';
+
+import { FormState, PrecomposedTransactionFinal } from '@suite-common/wallet-types';
+
+import * as modalActions from '../suite/modalActions';
+import * as metadataActions from 'src/actions/suite/metadataActions';
+import { Dispatch, GetState } from 'src/types/suite';
+import { MetadataAddPayload } from '@suite-common/metadata-types';
+
+import * as sendFormEthereumActions from './send/sendFormEthereumActions';
+
+// this could be called at any time during signTransaction or pushTransaction process (from TransactionReviewModal)
+export const cancelSignTx = () => (dispatch: Dispatch, getState: GetState) => {
+    const { signedTx } = getState().wallet.stake;
+    dispatch(stakeActions.requestSignTransaction());
+    dispatch(stakeActions.requestPushTransaction());
+    // if transaction is not signed yet interrupt signing in TrezorConnect
+    if (!signedTx) {
+        TrezorConnect.cancel('tx-cancelled');
+        return;
+    }
+    // otherwise just close modal
+    dispatch(modalActions.onCancel());
+};
+
+// private, called from signTransaction only
+const pushTransaction =
+    (signedTransaction: SignedTransaction['signedTransaction']) =>
+    async (dispatch: Dispatch, getState: GetState) => {
+        const { signedTx, precomposedTx } = getState().wallet.stake;
+        const { account } = getState().wallet.selectedAccount;
+        const device = selectDevice(getState());
+        if (!signedTx || !precomposedTx || !account) return;
+
+        const sentTx = await TrezorConnect.pushTransaction(signedTx);
+        // const sentTx = { success: true, payload: { txid: 'ABC ' } };
+
+        // close modal regardless result
+        dispatch(modalActions.onCancel());
+
+        const { token } = precomposedTx;
```

Comment:
token logic is not related to staking

Reply:
https://github.com/everstake/trezor-suite/pull/4/commits/2ed39aaff08c6509b0f48d27cef85971e157b43d